lib/api/CFieldDataTyper.cc

```diff
@@ -335,6 +335,15 @@ bool CFieldDataTyper::persistState(core::CDataAdder& persister) {
     return this->doPersistState(m_DataTyper->makePersistFunc(), m_ExamplesCollector, persister);
 }
 
+bool CFieldDataTyper::isPersistenceNeeded(const std::string& description) const {
+    if (m_NumRecordsHandled == 0) {
+        LOG_DEBUG(<< "Zero records were handled - will not attempt to persist "
+                  << description << ".");
+        return false;
```

Comment:
In the case of a categorizer chained to an anomaly detector this might not be correct - it's possible that time has been advanced on the anomaly detector.

So this method needs a `// Pass on the request in case we're chained` step like some of the other methods in this class have.  If the chained processor's `isPersistenceNeeded()` returns `true` then this method should return `true`, otherwise it should make its own decision using the code that's there now.